arduino-ide-extension/src/browser/theia/core/connection-status-service.ts

```diff
@@ -1,106 +1,324 @@
+  >();
+  private noInternetConnectionNotificationId: string | undefined;
+  private _offlineStatus: OfflineConnectionStatus | undefined;
 
-  protected connectedPort: string | undefined;
+  get offlineStatus(): OfflineConnectionStatus | undefined {
+    return this._offlineStatus;
+  }
 
-  @postConstruct()
-  protected async init(): Promise<void> {
-    try {
-      this.connectedPort = await this.daemon.tryGetPort();
-    } catch {}
-    this.notificationCenter.onDaemonDidStart(
-      (port) => (this.connectedPort = port)
-    );
-    this.notificationCenter.onDaemonDidStop(
-      () => (this.connectedPort = undefined)
-    );
+  get onOfflineStatusDidChange(): Event<OfflineConnectionStatus | undefined> {
+    return this.offlineStatusDidChangeEmitter.event;
   }
 
   protected override onStateChange(state: ConnectionStatus): void {
-    if (!this.connectedPort && state === ConnectionStatus.ONLINE) {
+    if (
+      (!Boolean(this.daemonPort.port) || !this.isOnline.online) &&
+      state === ConnectionStatus.ONLINE
+    ) {
       return;
     }
     super.onStateChange(state);
   }
 
   protected override handleOffline(): void {
-    this.statusBar.setElement('connection-status', {
+    const params = {
+      port: this.daemonPort.port,
+      online: this.isOnline.online,
+    };
+    this._offlineStatus = offlineConnectionStatusType(params);
+    const { text, tooltip } = offlineMessage(params);
+    this.statusBar.setElement(connectionStatusStatusBar, {
       alignment: StatusBarAlignment.LEFT,
-      text: this.connectedPort
-        ? nls.localize('theia/core/offline', 'Offline')
-        : '$(bolt) ' +
-          nls.localize('theia/core/daemonOffline', 'CLI Daemon Offline'),
-      tooltip: this.connectedPort
-        ? nls.localize(
-            'theia/core/cannotConnectBackend',
-            'Cannot connect to the backend.'
-          )
-        : nls.localize(
-            'theia/core/cannotConnectDaemon',
-            'Cannot connect to the CLI daemon.'
-          ),
+      text,
+      tooltip,
       priority: 5000,
     });
-    this.toDisposeOnOnline.push(
-      Disposable.create(() => this.statusBar.removeElement('connection-status'))
-    );
-    document.body.classList.add('theia-mod-offline');
-    this.toDisposeOnOnline.push(
+    document.body.classList.add(theiaOffline);
+    this.toDisposeOnOnline.pushAll([
       Disposable.create(() =>
-        document.body.classList.remove('theia-mod-offline')
-      )
-    );
+        this.statusBar.removeElement(connectionStatusStatusBar)
+      ),
+      Disposable.create(() => document.body.classList.remove(theiaOffline)),
+      Disposable.create(() => {
+        this._offlineStatus = undefined;
+        this.fireStatusDidChange();
+      }),
+    ]);
+    if (!this.isOnline.online) {
+      const text = nls.localize(
+        'arduino/connectionStatus/connectionLost',
+        "Connection lost. Cloud Sketches actions and updates won't be available."
```

Comment:
```suggestion
        "Connection lost. Cloud sketch actions and updates won't be available."
```

- "Sketches" -> "sketches": The decision was made that "sketch" is not a proper noun and thus should not be capitalized in sentence case text.
  reference: the "Arduino Editorial Guidelines" document in the ATL Confluence space
- "sketches" -> "sketch": Although the other isn't necessarily wrong, I think my suggestion is slightly cleaner grammatically.

Reply:
Thank you!

I decided to make the changes manually instead of committing the proposed change. If I commit this, I must still manually regenerate the `en.json` file with `yarn i18n:generate`.


https://user-images.githubusercontent.com/1405703/225549692-3d7e2871-70a5-45e4-8635-8aeb6ed45bfa.mp4